docs/source/processes.rst

```diff
@@ -770,13 +778,56 @@ described in |pywps-multi-output|_.
 .. seealso::
     - :ref:`Multiple and Optional Values`
 
+.. _exec_output_location:
+
+Outputs Location
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, :term:`Job` results will be hosted under the endpoint configured by ``weaver.wps_output_url`` and
+``weaver.wps_output_path``, and will be stored under directory defined by ``weaver.wps_output_dir`` setting.
+
+Each :term:`Job` will have its specific UUID employed for all of the outputs files, logs and status in order to
+avoid conflicts. Therefore, outputs will be available with the following location:
+
+.. code-block::
+
+    {WPS_OUTPUT_URL}/{JOB_UUID}.xml             # status location
+    {WPS_OUTPUT_URL}/{JOB_UUID}.log             # execution logs
+    {WPS_OUTPUT_URL}/{JOB_UUID}/{output.ext}    # results of the job if successful
+
+.. note::
+    Value ``WPS_OUTPUT_URL`` in above example is resolved accordingly with ``weaver.wps_output_url``,
+    ``weaver.wps_output_path`` and ``weaver.url``, as per :ref:`conf_settings` details.
+
+When submitting a :term:`Job` for execution, it is also possible to provide the ``X-WPS-Output-Context`` header.
+This modifies the output location to be nested under the specified directory or sub-directories.
+
+For example, providing ``X-WPS-Output-Context: project/test-1`` will result in outputs located at
+
+.. code-block::
+
+    {WPS_OUTPUT_URL}/project/test-1/{JOB_UUID}/{output.ext}
```

Comment:
In term of settings I'm not sure that we can avoid duplicating output_path/url defintion. Given the default output directory being the root of where I can specified sub-dir, I'm stuck to get that kind of folder:

- output_dir
    - uuid1.xml
    - uuid1.log
    - uuid1_dir
    - neat_workspace_buried_in_public_default_dir
        - user1
            - user1_uuid1.xml
            - ...  
        - user2
    - uuid2.xml
    - uuid2.log
    - uuid2_dir

So I see 2 choices:

1. We define a default output url/path and a context base output url/path that allow greater customization.
2. We add a default output context such as we could store default outputs to `{WPS_OUTPUT_URL}/{DEFAULT_OUTPUT_CONEXT}/{JOB_UUID}`. This way, the public stuff can be put deeper under the root directory.

Reply:
What I had in mind was to preserve `{DEFAULT_OUTPUT_CONEXT}` part as non existing by default to preserve the old behaviour on servers that don't require any context. In the case of DACCS for users use case, whether the a user is logged or not, a pre-hook in Twitcher/Magpie would always define the context. When not logged, that "context user" could be plainly `anonymous` or `public`. Therefore, the structure would be: 
```
output_dir
    anonymous
        uuid1.xml
        uuid1.logl
        uuid1_dir
    user1_id        
        uuid2.xml
        uuid2.logl
        uuid2_dir
```

Alternatively, I could define a `DEFAULT_OUTPUT_CONEXT` setting that is by default empty that can be overridden in the INI file specifically in birdhouse-deploy config.

Reply:
Given that Weaver (outside of DACCS context) offers an optional output context that can be set or not. I think it would make sense to provide that `DEFAULT_OUTPUT_CONEXT` option for deployments that would not garantee a context for each and every requests. Having an empty default value, make sense too to preserve the backward compatibility.

Reply:
See new commits for discussed default setting.